js/src/elements/star-rating.ts

```diff
@@ -238,18 +252,30 @@ export class StarRatingElement extends HTMLElement {
 		return parseInt(this.getAttribute('value') ?? '');
 	}
 
-	/** TODO: add form related properties (i.e validity, etc */
+	get validity() {
+		// borrow radio input's native validity
+		return this.shadowRoot?.querySelector('input')?.validity;
+	}
 
 	connectedCallback() {
 		this.shadowRoot?.addEventListener('change', this);
 		// arrow function to bind `this` value to star rating in handleFormData
-		this.closest('form')?.addEventListener('formdata', this);
+		this.form?.addEventListener('formdata', e => this.handleFormData(e));
 		this.shadowRoot?.addEventListener('slotchange', this);
+		this.setAttribute(
+			'aria-label',
+			this.querySelector('legend[slot="legend"]')
+				? (this.querySelector('legend[slot="legend"]') as HTMLLegendElement)?.innerText
+				: 'star rating'
+		);
+		if (!this.name) {
+			throw new Error(`${this.nodeName} id="${this.id}" requires a name attribute.`);
```

Comment:
do we need this or can we skip out in the formdata event when there's not a name? Normal inputs don't throw like this.

Reply:
Removed and it will not be included in formdata when there's no name.